build/patch/pre-autogen/libevent/common/000.patch

```diff
@@ -123,4 +94,26 @@ index 284341c..5b8b0a6 100644
  	    n>(size_t)INT_MAX ? INT_MAX : (int)n);
 +#endif
  }
--
+ 
+ void
+diff --git a/openssl-compat.h b/openssl-compat.h
+index 69afc71..8e3556a 100644
+--- a/openssl-compat.h
++++ b/openssl-compat.h
+@@ -1,7 +1,7 @@
+ #ifndef OPENSSL_COMPAT_H
+ #define OPENSSL_COMPAT_H
+ 
+-#if OPENSSL_VERSION_NUMBER < 0x10100000L
++#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined LIBRESSL_VERSION_NUMBER
+ 
+ static inline BIO_METHOD *BIO_meth_new(int type, const char *name)
+ {
+@@ -30,6 +30,6 @@ static inline BIO_METHOD *BIO_meth_new(int type, const char *name)
+ 
+ #define TLS_method SSLv23_method
+ 
+-#endif /* OPENSSL_VERSION_NUMBER < 0x10100000L */
++#endif /* OPENSSL_VERSION_NUMBER < 0x10100000L || defined LIBRESSL_VERSION_NUMBER */
```

Comment:
Basically the gist of it is that libressl uses an older API and here we enable backward compat for that.